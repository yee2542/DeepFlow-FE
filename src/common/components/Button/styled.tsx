import { ButtonComponentProps } from 'Components/Button/types';
import styled, { css } from 'styled-components';

const colorMapping = (color?: ButtonComponentProps['color']) => {
  switch (color) {
    case 'secondary':
      return css`
        background-color: #fff;
        color: var(--green-color);
        border: 1px solid var(--green-color);
        &:hover {
          color: var(--light-green-color);
          border: 1px solid var(--light-green-color);
        }
      `;
    case 'danger':
      return css`
        background-color: var(--error-color);
        color: #fff;
        &:hover {
          background-color: var(--light-error-color);
        }
      `;
    case 'success':
      return css`
        background-color: var(--success-color);
        color: #fff;
        &:hover {
          background-color: var(--light-success-color);
        }
      `;
    case 'disable':
      return css`
        background-color: var(--gray-color);
        color: var(--dark-gray-color);
        border: 1px solid var(--dark-gray-color);
        &:hover {
          cursor: not-allowed;
        }
      `;
    // default -> primary
    default:
      return css`
        background-color: var(--green-color);
        color: #fff;
        &:hover {
          background-color: var(--light-green-color);
        }
      `;
  }
};

const sizeMapping = (size?: ButtonComponentProps['size']) => {
  switch (size) {
    case 'small':
      return css`
        font-size: var(--small-body);
        padding: 10px 16px;
      `;
    case 'large':
      return css`
        font-size: var(--h5);
        padding: 12px 24px;
      `;
    // default -> medium
    default:
      return css`
        font-size: var(--body);
        padding: 11px 20px;
      `;
  }
};

export const ButtonStyle = styled.button<ButtonComponentProps>`
  border-radius: 100px;
  border: 0px;
  ${({ color }) => colorMapping(color)};
  ${({ size }) => sizeMapping(size)};
  cursor: pointer;
  font-weight: bold;
  &:focus {
    outline: 0;
    box-shadow: none !important;
  }
`;
