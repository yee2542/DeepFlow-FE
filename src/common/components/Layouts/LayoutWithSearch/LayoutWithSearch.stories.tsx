import { ApolloProvider } from '@apollo/client';
import { Meta, Story } from '@storybook/react';
import React from 'react';
import { Provider } from 'react-redux';
import { client } from 'Services/client';
import { store } from 'Stores/index';
import { GlobalStyle } from 'Styles/global';
import LayoutWithSearch from '.';

export default {
  title: 'FluentSearch/LayoutWithSearch',
  component: LayoutWithSearch,
} as Meta;

export const Default: Story = (): JSX.Element => (
  <>
    <GlobalStyle />
    <Provider store={store}>
      <ApolloProvider client={client}>
        <LayoutWithSearch title="Title"> Children here </LayoutWithSearch>
      </ApolloProvider>
    </Provider>
  </>
);
