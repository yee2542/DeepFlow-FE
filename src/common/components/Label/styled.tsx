import styled from 'styled-components';
import { Tag } from 'antd';

export const LabelStyle = styled(Tag)`
  border-radius: 100px;
  font-size: var(--small-body);
  padding: 4px 8px 4px 8px;
`;
