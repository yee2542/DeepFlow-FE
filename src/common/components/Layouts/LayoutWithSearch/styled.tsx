import styled, { css } from 'styled-components';
import { Layout } from 'antd';
const { Sider, Content } = Layout;

type ContentWithPaddingPropsType = {
  padding: boolean;
};

export const CanvasWrapper = styled.div`
  overflow: hidden;

  aside {
    width: 250px !important;
    max-width: 250px !important;
    min-width: 250px !important;
    flex: 0 0 250px !important;
  }

  .ant-layout-header {
    background-color: white;
    position: fixed;
    width: 100%;
    box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
    z-index: 999;
    /* margin-left: 10%; */
  }

  .ant-layout-content {
    background-color: white;
    overflow-y: scroll;
    width: 100%;

    @media (min-width: 1600px) {
      padding: 10% 5% 5% 10%;
    }
    @media (min-width: 1200px) and (max-width: 1599px) {
      padding: 10% 5% 5% 15%;
    }
    @media (min-width: 992px) and (max-width: 1199px) {
      padding: 10% 5% 5% 20%;
    }
    @media (max-width: 991px) {
      padding: 20% 10% 5% 10%;
    }
  }
`;

export const ContentWrapper = styled(Layout)`
  margin-left: 10%;

  @media (max-width: 992px) {
    margin-left: 0%;
  }
`;

export const SiderWrapper = styled(Sider)`
  background-color: #4425a7;
  padding: 5% 2% 0% 2%;
  overflow: hidden;
  z-index: 1000;
  left: 0;
  position: fixed;
  height: 100vh;

  @media (max-width: 992px) {
    display: none;
  }
`;

export const MenuWrapper = styled.div`
  margin-top: 30%;
  width: 100%;
`;

export const IconWrapper = styled.span`
  margin-right: 10%;
`;

export const MenuItem = styled.div<{ isSelecting: boolean }>`
  width: 100%;
  cursor: pointer;
  color: white;

  p:hover {
    padding: 1em 1em 1em 1em;
    border-radius: 0.5em;
    background-color: #48cdd8;
  }

  p {
    padding: 1em 1em 1em 1em;
    border-radius: 0.5em;
    background-color: transparent;
  }

  ${(props) =>
    props.isSelecting &&
    css`
      p {
        padding: 1em 1em 1em 1em;
        border-radius: 0.5em;
        background-color: white;
        color: #4425a7;
        font-weight: 600;
      }

      p:hover {
        padding: 1em 1em 1em 1em;
        border-radius: 0.5em;
        background-color: #9271fc;
      }
    `}
`;

export const WrapperImage = styled.section`
  max-width: 70rem;
  margin: 4rem auto;
  display: grid;
  grid-gap: 1em;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  grid-auto-rows: 300px;
`;

export const UserWrapper = styled.div`
  cursor: default;

  .ant-avatar-circle {
    color: #fde3cf;
    background-color: #f56a00;
  }
  .ant-avatar-circle :hover {
    color: #f56a00;
    background-color: #fde3cf;
  }
`;

export const ContentWithPadding = styled(Content)<ContentWithPaddingPropsType>`
  ${(props) =>
    props.padding &&
    `
    padding: 5% 0% 5% 0%;
    margin-left: 5%;
  `}
`;
