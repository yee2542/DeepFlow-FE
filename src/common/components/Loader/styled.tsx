import styled from 'styled-components';

export const Loading = styled.div`
  text-align: center;
  .anticon {
    font-size: 24px;
    color: #4425a7;
  }
`;
