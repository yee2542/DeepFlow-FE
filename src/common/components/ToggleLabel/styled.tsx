import styled from 'styled-components';
import { Tag } from 'antd';
import { ToggleLabelWrapperProps } from './types';

export const ToggelLabelWrapper = styled(Tag)<ToggleLabelWrapperProps>`
  border-radius: 100px;
  font-size: var(--small-body);
  padding: 0em 1.25em 0em 1.25em;
  background-color: white;
  color: #5a36cc;
  border: 1px solid #5a36cc;
  ${(props) =>
    props.selected &&
    `background-color: #5A36CC;
    color: white;
    `}
  &:hover {
    background-color: #f8eafa;
    color: #5a36cc;
    cursor: pointer;
  }
`;
