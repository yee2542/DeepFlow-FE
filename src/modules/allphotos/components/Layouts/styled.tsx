import styled from 'styled-components';
import { Layout } from 'antd';
const { Sider } = Layout;

export const CanvasWrapper = styled.div`
  .ant-layout-header {
    background-color: white;
  }

  .ant-layout-content {
    height: 100vh;
    background-color: white;
  }
`;

export const SiderWrapper = styled(Sider)`
  background-color: #4425a7;
  width: 294px;
  padding: 2%;
`;

export const MenuWrapper = styled.div`
  margin-top: 20%;
  width: 100%;
`;

export const Menu = styled.div`
  margin-top: 20%;
  a:hover {
    padding: 10%;
    border-radius: 2em;
    background-color: #48cdd8;
  }
  a:active {
    padding: 10%;
    border-radius: 2em;
    background-color: #48cdd8;
    position: inherit;
  }
  a {
    color: white;
  }
  width: 100%;
`;
