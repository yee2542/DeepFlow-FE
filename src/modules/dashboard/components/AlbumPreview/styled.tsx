import styled from 'styled-components';
import { Tag } from 'antd';

export const AlbumWrapper = styled.div`
  img {
    border-radius: 8px;
    opacity: 1;
    display: block;
    width: 100%;
    height: auto;
    transition: 0.5s ease;
    backface-visibility: hidden;
  }
`;

export const Label = styled(Tag)`
  border-radius: 100px;
  font-size: 0.8rem !important;
`;

export const Container = styled.div`
  position: relative;

  &:hover img {
    opacity: 0.3;
    cursor: pointer;
  }

  &:hover #text {
    cursor: pointer;
    opacity: 1;
  }
`;

export const Text = styled.div`
  color: white;
  font-size: 20px;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  text-align: center;
  opacity: 0;
`;
