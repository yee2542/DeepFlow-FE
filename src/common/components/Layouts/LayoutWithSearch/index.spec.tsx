import { shallow } from 'enzyme';
import React from 'react';
import LayoutWithSearch from '.';
import { ContentWithPadding } from './styled';

describe('Component/LayoutWithSearch test', () => {
  it('Should match snapshots', () => {
    const wrap = shallow(<LayoutWithSearch />);
    expect(wrap).toMatchSnapshot();
  });

  it('It should have title and children', () => {
    const wrapper = shallow(
      <LayoutWithSearch title="Title"> Children here </LayoutWithSearch>,
    );

    expect(wrapper.exists(ContentWithPadding)).toBe(true);
    expect(wrapper.find(ContentWithPadding).text()).toBe('Title Children here ');
    expect(wrapper.exists('hr')).toBe(true);
    expect(wrapper.find('h1').text()).toBe('Title');
  });

  it('It should not have title and children', () => {
    const wrapper = shallow(<LayoutWithSearch />);

    expect(wrapper.exists(ContentWithPadding)).toBe(true);
    expect(wrapper.find(ContentWithPadding).text()).toBe('');
    expect(wrapper.exists('hr')).toBe(true);
    expect(wrapper.find('h1').text()).toBe('');
  });
});
