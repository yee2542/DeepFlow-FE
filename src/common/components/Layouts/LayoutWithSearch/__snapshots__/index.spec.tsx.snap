// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/LayoutWithSearch test Should match snapshots 1`] = `ShallowWrapper {}`;
